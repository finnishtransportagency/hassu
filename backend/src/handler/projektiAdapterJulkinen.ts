import {
  AloitusKuulutusJulkaisu,
  AloitusKuulutusPDF,
  DBProjekti,
  LocalizedMap,
  SuunnitteluSopimus,
} from "../database/model/projekti";
import * as API from "../../../common/graphql/apiModel";
import pickBy from "lodash/pickBy";
import dayjs from "dayjs";
import {
  adaptAineistot,
  adaptHankkeenKuvaus,
  adaptKielitiedot,
  adaptLinkkiList,
  adaptVelho,
  adaptVuorovaikutusTilaisuudet,
  adaptYhteystiedot,
} from "./projektiAdapter";
import { fileService } from "../files/fileService";
import { log } from "../logger";
import { parseDate } from "../util/dateUtil";
import { Vuorovaikutus } from "../database/model/suunnitteluVaihe";

class ProjektiAdapterJulkinen {
  applyStatus(projekti: API.ProjektiJulkinen) {
    function checkAloituskuulutus() {
      if (projekti.aloitusKuulutusJulkaisut) {
        const julkisetAloituskuulutukset = projekti.aloitusKuulutusJulkaisut.filter((julkaisu) => {
          return julkaisu.kuulutusPaiva && parseDate(julkaisu.kuulutusPaiva).isBefore(dayjs());
        });

        if (julkisetAloituskuulutukset?.length > 0) {
          projekti.status = API.Status.ALOITUSKUULUTUS;
        }
      }
    }

    function checkSuunnittelu() {
      // Valiaikainen ui kehitysta varten, kunnes suunnitteluvaihe tietomallissa
      if (projekti.suunnitteluVaihe) {
        projekti.status = API.Status.SUUNNITTELU;
      }
    }

    projekti.status = API.Status.EI_JULKAISTU;

    checkAloituskuulutus();

    checkSuunnittelu();

    // checkNahtavillaolo();

    // checkHyvaksyttavana();

    // checkPaatos();

    // checkLainvoima();

    return projekti;
  }

  public adaptProjekti(dbProjekti: DBProjekti): API.ProjektiJulkinen | undefined {
    const aloitusKuulutusJulkaisut = this.adaptAloitusKuulutusJulkaisut(
      dbProjekti.oid,
      dbProjekti.aloitusKuulutusJulkaisut
    );

    if (!checkIfAloitusKuulutusJulkaisutIsPublic(aloitusKuulutusJulkaisut)) {
      return undefined;
    }

    let suunnitteluVaihe = undefined;
    if (dbProjekti.suunnitteluVaihe?.julkinen) {
      suunnitteluVaihe = ProjektiAdapterJulkinen.adaptSuunnitteluVaihe(dbProjekti);
    }

    const projekti: API.ProjektiJulkinen = {
      __typename: "ProjektiJulkinen",
      oid: dbProjekti.oid,
      euRahoitus: dbProjekti.euRahoitus,
      suunnitteluVaihe,
      aloitusKuulutusJulkaisut,
    };
    return removeUndefinedFields(projekti) as API.ProjektiJulkinen;
  }

  adaptAloitusKuulutusJulkaisut(
    oid: string,
    aloitusKuulutusJulkaisut?: AloitusKuulutusJulkaisu[] | null
  ): API.AloitusKuulutusJulkaisuJulkinen[] | undefined {
    if (aloitusKuulutusJulkaisut) {
      // Pick HYVAKSYTTY or MIGROITU aloituskuulutusjulkaisu, by this order
      const julkaisu =
        findJulkaisuByStatus(aloitusKuulutusJulkaisut, API.AloitusKuulutusTila.HYVAKSYTTY) ||
        findJulkaisuByStatus(aloitusKuulutusJulkaisut, API.AloitusKuulutusTila.MIGROITU);
      const { yhteystiedot, velho, suunnitteluSopimus, kielitiedot } = julkaisu;

      return [
        {
          __typename: "AloitusKuulutusJulkaisuJulkinen",
          kuulutusPaiva: julkaisu.kuulutusPaiva,
          siirtyySuunnitteluVaiheeseen: julkaisu.siirtyySuunnitteluVaiheeseen,
          hankkeenKuvaus: adaptHankkeenKuvaus(julkaisu.hankkeenKuvaus),
          yhteystiedot: adaptYhteystiedot(yhteystiedot),
          velho: adaptVelho(velho),
          suunnitteluSopimus: this.adaptSuunnitteluSopimus(oid, suunnitteluSopimus),
          kielitiedot: adaptKielitiedot(kielitiedot),
          aloituskuulutusPDFt: this.adaptJulkaisuPDFPaths(oid, julkaisu.aloituskuulutusPDFt),
          tila: julkaisu.tila,
        },
      ];
    }
    return undefined;
  }

  adaptSuunnitteluSopimus(
    oid: string,
    suunnitteluSopimus?: SuunnitteluSopimus | null
  ): API.SuunnitteluSopimus | undefined | null {
    if (suunnitteluSopimus) {
      return {
        __typename: "SuunnitteluSopimus",
        ...suunnitteluSopimus,
        logo: fileService.getPublicPathForProjektiFile(oid, suunnitteluSopimus.logo),
      };
    }
    return suunnitteluSopimus as undefined | null;
  }

  adaptJulkaisuPDFPaths(
    oid: string,
    aloitusKuulutusPDFS: LocalizedMap<AloitusKuulutusPDF>
  ): API.AloitusKuulutusPDFt | undefined {
    if (!aloitusKuulutusPDFS) {
      return undefined;
    }

    const result = {};
    for (const kieli in aloitusKuulutusPDFS) {
      result[kieli] = {
        aloituskuulutusPDFPath: fileService.getPublicPathForProjektiFile(
          oid,
          aloitusKuulutusPDFS[kieli].aloituskuulutusPDFPath
        ),
        aloituskuulutusIlmoitusPDFPath: fileService.getPublicPathForProjektiFile(
          oid,
          aloitusKuulutusPDFS[kieli].aloituskuulutusIlmoitusPDFPath
        ),
      } as AloitusKuulutusPDF;
    }
    return { __typename: "AloitusKuulutusPDFt", SUOMI: result[API.Kieli.SUOMI], ...result };
  }

  private static adaptSuunnitteluVaihe(dbProjekti: DBProjekti): API.SuunnitteluVaiheJulkinen {
    const { hankkeenKuvaus, arvioSeuraavanVaiheenAlkamisesta, suunnittelunEteneminenJaKesto } =
      dbProjekti.suunnitteluVaihe;
    return {
      __typename: "SuunnitteluVaiheJulkinen",
      hankkeenKuvaus: adaptHankkeenKuvaus(hankkeenKuvaus),
      arvioSeuraavanVaiheenAlkamisesta,
      suunnittelunEteneminenJaKesto,
      vuorovaikutukset: adaptVuorovaikutukset(dbProjekti),
    };
  }
}

function adaptVuorovaikutukset(dbProjekti: DBProjekti): API.VuorovaikutusJulkinen[] {
  const vuorovaikutukset = dbProjekti.vuorovaikutukset;
  if (vuorovaikutukset && vuorovaikutukset.length > 0) {
    return vuorovaikutukset
      .map((vuorovaikutus) => {
        const julkaisuPaiva = parseDate(vuorovaikutus.vuorovaikutusJulkaisuPaiva);
        if (julkaisuPaiva.isBefore(dayjs())) {
          return {
            ...vuorovaikutus,
            __typename: "VuorovaikutusJulkinen",
            vuorovaikutusTilaisuudet: adaptVuorovaikutusTilaisuudet(vuorovaikutus.vuorovaikutusTilaisuudet),
            videot: adaptLinkkiList(vuorovaikutus.videot),
            aineistot: adaptAineistot(vuorovaikutus.aineistot, julkaisuPaiva),
            vuorovaikutusYhteystiedot: adaptAndMergeYhteystiedot(dbProjekti, vuorovaikutus),
          } as API.VuorovaikutusJulkinen;
        }
        return undefined;
      })
      .filter((obj) => obj);
  }
  return vuorovaikutukset as undefined;
}

function findJulkaisuByStatus<T extends { tila?: API.AloitusKuulutusTila }>(
  aloitusKuulutusJulkaisut: T[],
  tila: API.AloitusKuulutusTila
): T {
  return aloitusKuulutusJulkaisut.filter((j) => j.tila == tila).pop();
}

function checkIfAloitusKuulutusJulkaisutIsPublic(
  aloitusKuulutusJulkaisut: API.AloitusKuulutusJulkaisuJulkinen[]
): boolean {
  if (!(aloitusKuulutusJulkaisut && aloitusKuulutusJulkaisut.length == 1)) {
    log.info("Projektilla ei ole hyv채ksytty채 aloituskuulutusta");
    return false;
  }

  const hyvaksyttyJulkaisu = findJulkaisuByStatus(aloitusKuulutusJulkaisut, API.AloitusKuulutusTila.HYVAKSYTTY);
  if (hyvaksyttyJulkaisu) {
    if (hyvaksyttyJulkaisu.kuulutusPaiva && parseDate(hyvaksyttyJulkaisu.kuulutusPaiva).isAfter(dayjs())) {
      log.info("Projektin aloituskuulutuksen kuulutusp채iv채 on tulevaisuudessa", {
        kuulutusPaiva: parseDate(hyvaksyttyJulkaisu.kuulutusPaiva).format(),
        now: dayjs().format(),
      });
      return false;
    }
  } else if (!findJulkaisuByStatus(aloitusKuulutusJulkaisut, API.AloitusKuulutusTila.MIGROITU)) {
    // If there are no HYVAKSYTTY or MIGROITU aloitusKuulutusJulkaisu, hide projekti
    return false;
  }
  return true;
}

function adaptAndMergeYhteystiedot(dbProjekti: DBProjekti, vuorovaikutus: Vuorovaikutus) {
  let vuorovaikutusYhteystiedot = adaptYhteystiedotFromUsernames(dbProjekti, vuorovaikutus.vuorovaikutusYhteysHenkilot);
  if (!vuorovaikutusYhteystiedot) {
    vuorovaikutusYhteystiedot = [];
  }
  const yhteystiedot = adaptYhteystiedot(vuorovaikutus.esitettavatYhteystiedot);
  if (yhteystiedot) {
    vuorovaikutusYhteystiedot = vuorovaikutusYhteystiedot.concat(yhteystiedot);
  }
  return vuorovaikutusYhteystiedot;
}

function adaptYhteystiedotFromUsernames(
  dbProjekti: DBProjekti,
  usernames?: Array<string>
): API.Yhteystieto[] | undefined {
  if (!usernames || usernames.length == 0) {
    return undefined;
  }
  const kayttoOikeudet = dbProjekti.kayttoOikeudet;
  return usernames
    .map((username) => {
      const user = kayttoOikeudet.find((projektiUser) => projektiUser.kayttajatunnus == username);
      if (!user) {
        return undefined;
      }
      const lastnameFirstname = user.nimi.split(",");
      return {
        __typename: "Yhteystieto",
        etunimi: lastnameFirstname[1]?.trim(),
        sukunimi: lastnameFirstname[0]?.trim(),
        organisaatio: user.organisaatio,
        puhelinnumero: user.puhelinnumero,
        sahkoposti: user.email,
      } as API.Yhteystieto;
    })
    .filter((obj) => obj);
}

function removeUndefinedFields(object: API.ProjektiJulkinen): Partial<API.ProjektiJulkinen> {
  return pickBy(object, (value) => value !== undefined);
}

export const projektiAdapterJulkinen = new ProjektiAdapterJulkinen();
