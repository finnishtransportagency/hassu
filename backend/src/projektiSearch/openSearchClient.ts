import { HttpRequest } from "@aws-sdk/protocol-http";
import { openSearchConfig } from "./openSearchConfig";
import { sendSignedRequest } from "../aws/awsRequest";
import { log } from "../logger";

const domain = openSearchConfig.searchDomain ?? "search-domain-missing";
const type = "_doc";

export async function sendRequest(request: HttpRequest): Promise<unknown> {
  return (await sendSignedRequest(request, "es")).body;
}

export interface SortOrder {
  order: "asc" | "desc";
}

export enum OpenSearchIndexType {
  YLLAPITO = "yllapito",
  JULKINEN = "julkinen",
}

export interface SearchOpts {
  from?: number;
  size?: number;
  sort?: Record<string, SortOrder>[];
  query?: unknown;
  aggs?: unknown;
}

export default class OpenSearchClient {
  private index: string;

  constructor(index: string) {
    this.index = index;
  }

  async putDocument(key: string, document: unknown): Promise<void> {
    const response = (await this.put("/" + type + "/" + key, JSON.stringify(document))) as { result: string };
    if (response["result"] !== "created" && response["result"] !== "updated") {
      log.warn(response);
    }
  }

  async deleteDocument(key: string): Promise<void> {
    await this.delete("/" + type + "/" + key);
  }

  async delete(path: string): Promise<unknown> {
    const request = new HttpRequest({
      headers: {
        host: domain,
      },
      hostname: domain,
      method: "DELETE",
      path: this.index + path,
    });
    return sendRequest(request);
  }

  async deleteIndex(): Promise<unknown> {
    return this.delete("");
  }

  async createIndex(settings: any, mappings: any): Promise<unknown> {
    const body = JSON.stringify({ settings, mappings });
    return this.put("", body);
  }

  async indexExists(): Promise<boolean> {
    const request = new HttpRequest({
      headers: {
        host: domain,
      },
      query: { pretty: "true" },
      hostname: domain,
      method: "HEAD",
      path: this.index,
    });
    try {
      const response = await sendSignedRequest(request, "es");
      return response.statusCode === 200;
    } catch (e) {
      log.info("No index found", e);
    }
    return false;
  }

  async query(query: SearchOpts): Promise<any> {
    const body = JSON.stringify(query);
    log.info("query " + this.index, { query });
    const request = new HttpRequest({
      body,
      headers: {
        "Content-Type": "application/json",
        host: domain,
      },
      hostname: domain,
      method: "POST",
      path: this.index + "/_search",
    });
    return sendRequest(request);
  }

  async getSettings(): Promise<unknown> {
    return this.get("/_settings");
  }

  async putSettings(settings: string): Promise<unknown> {
    await this.closeIndex();

    try {
      return await this.put("/_settings", settings);
    } finally {
      await this.openIndex();
    }
  }

  async getMapping(): Promise<unknown> {
    const mapping = await this.get("/_mapping");
    return (mapping as any).projekti.mappings;
  }

  async putMapping(mapping: string): Promise<unknown> {
    return this.put("/_mapping", mapping);
  }

  async get(path: string): Promise<any> {
    const request = new HttpRequest({
      headers: {
        host: domain,
      },
      query: { pretty: "true" },
      hostname: domain,
      method: "GET",
      path: this.index + path,
    });
    return sendRequest(request);
  }

  async put(path: string, body: string): Promise<unknown> {
    const request = new HttpRequest({
      headers: {
        "Content-Type": "application/json",
        host: domain,
      },
      body,
      hostname: domain,
      method: "PUT",
      path: this.index + path,
    });
    return sendRequest(request);
  }

  protected async closeIndex(): Promise<void> {
    log.info("closeIndex " + this.index);
    const request = new HttpRequest({
      headers: {
        "Content-Type": "application/json",
        host: domain,
      },
      hostname: domain,
      method: "POST",
      path: this.index + "/_close",
    });
    log.info(await sendRequest(request));
  }

  private async openIndex() {
    log.info("openIndex " + this.index);
    const request = new HttpRequest({
      headers: {
        "Content-Type": "application/json",
        host: domain,
      },
      hostname: domain,
      method: "POST",
      path: this.index + "/_open",
    });
    log.info(sendRequest(request));
  }
}
