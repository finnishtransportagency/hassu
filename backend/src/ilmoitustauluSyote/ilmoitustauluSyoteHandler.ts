import { Kieli } from "../../../common/graphql/apiModel";
import RSS from "rss";
import { openSearchClientIlmoitustauluSyote } from "../projektiSearch/openSearchClient";
import { IlmoitusKuulutus } from "./ilmoitusKuulutus";
import { translate } from "../util/localization";
import { kuntametadata } from "../../../common/kuntametadata";
import { log } from "../logger";
import { NotFoundError } from "../error/NotFoundError";
import { KaannettavaKieli } from "../../../common/kaannettavatKielet";

class IlmoitustauluSyoteHandler {
  async getFeed(kieli: KaannettavaKieli, ely: string | undefined, lely: string | undefined, maakunta: string | undefined): Promise<string> {
    const siteUrl = process.env.FRONTEND_DOMAIN_NAME || "";
    const feed_url = siteUrl + "/api/kuulutukset";
    const feed = new RSS({ feed_url, site_url: siteUrl, title: "Kuulutukset" });

    const terms: unknown[] =
      kieli === Kieli.SUOMI
        ? [
            {
              terms: { "kieli.keyword": [kieli, Kieli.POHJOISSAAME] },
            },
          ]
        : [
            {
              term: { "kieli.keyword": kieli },
            },
          ];
    const elyId = ely ? kuntametadata.elyIdFromKey(ely) : undefined;
    if (ely) {
      if (elyId) {
        terms.push({
          term: { "elyt.keyword": elyId },
        });
      } else {
        throw new NotFoundError("ELY " + ely + " on tuntematon");
      }
    }

    const lelyId = lely ? kuntametadata.elyIdFromKey(lely) : undefined;
    if (lely) {
      if (lelyId) {
        terms.push({
          term: { "lelyt.keyword": lelyId },
        });
      } else {
        throw new NotFoundError("Liikenne-ELY " + lely + " on tuntematon");
      }
    }

    if (maakunta) {
      try {
        const maakuntaId = kuntametadata.idForMaakuntaName(maakunta);
        terms.push({
          term: { maakunnat: maakuntaId },
        });
      } catch (e) {
        log.warn(e);
        throw new NotFoundError("Maakunta " + maakunta + " on tuntematon");
      }
    }
    const searchResult = await openSearchClientIlmoitustauluSyote.query({
      query: {
        bool: {
          must: terms,
        },
      },
      sort: [{ date: { order: "desc" } }],
    });

    if (searchResult.hits?.hits) {
      for (const item of searchResult.hits?.hits) {
        const kuulutus = item._source as IlmoitusKuulutus;
        const categories = this.getCategories(kuulutus);
        const date = kuulutus.date;
        const description = "";
        const title = kuulutus.title;
        const url = kuulutus.url;
        feed.item({ guid: item._id, categories, date, description, title, url });
      }
    }
    return feed.xml({ indent: true });
  }

  public getCategories(item: IlmoitusKuulutus) {
    const categories: string[] = ["Kuulutukset ja ilmoitukset:Kuulutus"];
    item.kunnat?.forEach((kuntaId) => {
      const kunta = kuntametadata.nameForKuntaId(kuntaId, Kieli.SUOMI);
      categories.push("Kunta:" + kunta);
    });
    item.maakunnat?.forEach((maakuntaId) => {
      const maakunta = kuntametadata.nameForMaakuntaId(maakuntaId, Kieli.SUOMI);
      categories.push("Maakunta:" + maakunta);
    });
    item.vaylamuoto?.forEach((vaylamuoto) => {
      const translatedVaylamuoto = translate("projekti.projekti-vayla-muoto." + vaylamuoto, Kieli.SUOMI);
      if (translatedVaylamuoto) {
        categories.push("Väylämuoto:" + translatedVaylamuoto);
      }
    });
    return categories;
  }
}

export const ilmoitustauluSyoteHandler = new IlmoitustauluSyoteHandler();
