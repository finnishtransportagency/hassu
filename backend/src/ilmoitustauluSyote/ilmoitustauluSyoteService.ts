import { AloitusKuulutusTila, ProjektiJulkinen } from "../../../common/graphql/apiModel";
import { openSearchClientIlmoitustauluSyote } from "../projektiSearch/openSearchClient";
import { ilmoitusKuulutusAdapter } from "./ilmoitustauluSyoteAdapter";

class IlmoitustauluSyoteService {
  async index(projekti: ProjektiJulkinen) {
    const oid = projekti.oid;
    const kielet = ilmoitusKuulutusAdapter.getProjektiKielet(projekti);

    const aloitusKuulutusJulkaisut = projekti.aloitusKuulutusJulkaisut?.filter(
      (julkaisu) => julkaisu.tila == AloitusKuulutusTila.HYVAKSYTTY
    );
    if (aloitusKuulutusJulkaisut) {
      for (const aloitusKuulutusJulkaisu of aloitusKuulutusJulkaisut) {
        for (const kieli of kielet) {
          await openSearchClientIlmoitustauluSyote.putDocument(
            ilmoitusKuulutusAdapter.createKeyForAloitusKuulutusJulkaisu(oid, aloitusKuulutusJulkaisu, kieli),
            ilmoitusKuulutusAdapter.adaptAloitusKuulutusJulkaisu(oid, aloitusKuulutusJulkaisu, kieli)
          );
        }
      }
    }
  }

  async remove(oid: string) {
    const searchResult = await openSearchClientIlmoitustauluSyote.query({
      query: {
        bool: {
          must: [
            {
              term: { "oid.keyword": oid },
            },
          ],
        },
      },
    });

    if (searchResult.hits?.hits) {
      for (const item of searchResult.hits?.hits) {
        await openSearchClientIlmoitustauluSyote.deleteDocument(item._id);
      }
    }
  }
}

export const ilmoitustauluSyoteService = new IlmoitustauluSyoteService();
