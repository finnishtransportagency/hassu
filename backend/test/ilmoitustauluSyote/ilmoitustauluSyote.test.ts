import { describe, it } from "mocha";
import sinon from "sinon";
import { ilmoitustauluSyoteService } from "../../src/ilmoitustauluSyote/ilmoitustauluSyoteService";
import { ProjektiFixture } from "../fixture/projektiFixture";
import { AloitusKuulutusTila, ProjektiJulkinen } from "../../../common/graphql/apiModel";
import { projektiAdapterJulkinen } from "../../src/projekti/adapter/projektiAdapterJulkinen";
import { openSearchClientIlmoitustauluSyote } from "../../src/projektiSearch/openSearchClient";
import { ilmoitustauluSyoteHandler } from "../../src/ilmoitustauluSyote/ilmoitustauluSyoteHandler";

const { expect } = require("chai");
const sandbox = sinon.createSandbox();

describe("IlmoitustauluSyote", () => {
  let putDocumentStub: sinon.SinonStub;
  let projekti: ProjektiJulkinen;

  beforeEach(() => {
    putDocumentStub = sandbox.stub(openSearchClientIlmoitustauluSyote, "putDocument");
    const projektiFixture = new ProjektiFixture();
    projekti = projektiAdapterJulkinen.adaptProjekti(projektiFixture.dbProjekti4())!;
    expect(projekti.aloitusKuulutusJulkaisu?.tila).to.eql(AloitusKuulutusTila.HYVAKSYTTY);
  });

  afterEach(() => {
    sandbox.reset();
  });

  after(() => {
    sandbox.restore();
  });

  it("Should index projekti into ilmoitustaulusyote index successfully", async () => {
    await ilmoitustauluSyoteService.index(projekti);
    expect(putDocumentStub.getCalls().map((call) => call.lastArg)).toMatchSnapshot();

    const indexedAloitusKuulutus = putDocumentStub!.getCalls()?.slice(0, 1)?.pop()?.lastArg;
    expect(ilmoitustauluSyoteHandler.getCategories(indexedAloitusKuulutus)).to.eql([
      "Kuulutukset ja ilmoitukset:Kuulutus",
      "Kunta:Mikkeli",
      "Kunta:Juva",
      "Kunta:Savonlinna",
      "Maakunta:Uusimaa",
      "Maakunta:Pirkanmaa",
    ]);
  });
});
