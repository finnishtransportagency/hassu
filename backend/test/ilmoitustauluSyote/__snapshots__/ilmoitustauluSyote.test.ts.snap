// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IlmoitustauluSyote Should index projekti into ilmoitustaulusyote index successfully 1`] = `
Array [
  Object {
    "date": "2022-03-28T14:28:00+03:00",
    "elyt": Array [
      "ely/ely07",
    ],
    "kieli": "SUOMI",
    "kunnat": Array [
      491,
      178,
      740,
    ],
    "lelyt": Array [
      "ely/ely08",
      "ely/ely08",
      "ely/ely08",
    ],
    "maakunnat": Array [
      1,
      6,
    ],
    "oid": "4",
    "title": "Kuulutus suunnittelun aloittamisesta: Marikan testiprojekti",
    "type": "Kuulutus",
    "url": "https://localhost:3000/suunnitelma/4/aloituskuulutus",
    "vaylamuoto": Array [
      "tie",
    ],
  },
  Object {
    "date": "2022-03-28T14:28:00+03:00",
    "elyt": Array [
      "ely/ely07",
    ],
    "kieli": "RUOTSI",
    "kunnat": Array [
      491,
      178,
      740,
    ],
    "lelyt": Array [
      "ely/ely08",
      "ely/ely08",
      "ely/ely08",
    ],
    "maakunnat": Array [
      1,
      6,
    ],
    "oid": "4",
    "title": "Kungörelse om inledande av planering: Marikas testprojekt",
    "type": "Kuulutus",
    "url": "https://localhost:3000/sv/suunnitelma/4/aloituskuulutus",
    "vaylamuoto": Array [
      "tie",
    ],
  },
  Object {
    "date": "2022-06-20T11:54:00+03:00",
    "elyt": Array [
      "ely/ely01",
    ],
    "kieli": "SUOMI",
    "kunnat": Array [
      245,
    ],
    "lelyt": Array [
      "ely/ely01",
    ],
    "maakunnat": Array [
      1,
      6,
    ],
    "oid": "4",
    "title": "Kuulutus suunnitelman nähtäville asettamisesta: Mt 140 parantaminen Kaskelantien kohdalla, tiesuunnitelma, Kerava",
    "type": "Kuulutus",
    "url": "https://localhost:3000/suunnitelma/4/suunnittelu",
    "vaylamuoto": Array [
      "tie",
    ],
  },
  Object {
    "date": "2022-06-20T11:54:00+03:00",
    "elyt": Array [
      "ely/ely01",
    ],
    "kieli": "RUOTSI",
    "kunnat": Array [
      245,
    ],
    "lelyt": Array [
      "ely/ely01",
    ],
    "maakunnat": Array [
      1,
      6,
    ],
    "oid": "4",
    "title": "Kungörelse om framläggande av plan: sv",
    "type": "Kuulutus",
    "url": "https://localhost:3000/sv/suunnitelma/4/suunnittelu",
    "vaylamuoto": Array [
      "tie",
    ],
  },
  Object {
    "date": "2022-06-09T00:00:00+03:00",
    "elyt": Array [
      "ely/ely01",
    ],
    "kieli": "SUOMI",
    "kunnat": Array [
      91,
      92,
    ],
    "lelyt": Array [
      "ely/ely01",
      "ely/ely01",
    ],
    "maakunnat": Array [
      1,
    ],
    "oid": "4",
    "title": "Kuulutus suunnitelman hyväksymispäätöksestä: HASSU AUTOMAATTITESTIPROJEKTI1",
    "type": "Kuulutus",
    "url": "https://localhost:3000/suunnitelma/4/hyvaksymispaatos",
    "vaylamuoto": Array [
      "tie",
    ],
  },
  Object {
    "date": "2022-06-09T00:00:00+03:00",
    "elyt": Array [
      "ely/ely01",
    ],
    "kieli": "RUOTSI",
    "kunnat": Array [
      91,
      92,
    ],
    "lelyt": Array [
      "ely/ely01",
      "ely/ely01",
    ],
    "maakunnat": Array [
      1,
    ],
    "oid": "4",
    "title": "Kungörelse av beslutet att godkänna planen: Namnet på svenska",
    "type": "Kuulutus",
    "url": "https://localhost:3000/sv/suunnitelma/4/hyvaksymispaatos",
    "vaylamuoto": Array [
      "tie",
    ],
  },
]
`;
